Editor/CinemachineEditorAnalytics.cs

```diff
@@ -0,0 +1,286 @@
+                            case "CinemachineBasicMultiChannelPerlin":
+                                noiseComponent = type;
+                                break;
+                            default:
+                                noiseComponent = k_customStr; // hide users component's name
+                                customComponentCount++;
+                                break;
+                        }
+
+                        break;
+                    default:
+                        customComponentCount++;
+                        break;
+                }
+            }
+        }
+
+        return new VcamData
+        {
+            id = id,
+            vcam_class = GetVcamClassName(vcam),
+            has_follow_target = vcam.Follow != null,
+            has_lookat_target = vcam.LookAt != null,
+            blend_hint = vcam.m_Transitions.m_BlendHint.ToString(),
+            inherit_position = vcam.m_Transitions.m_InheritPosition,
+            stand_by_update = vcam.m_StandbyUpdate.ToString(),
+            mode_overwrite = vcam.m_Lens.ModeOverride.ToString(),
+            body_component = bodyComponent,
+            aim_component = aimComponent,
+            noise_component = noiseComponent,
+            custom_component_count = customComponentCount,
+            extensions = vcamExtensions.ToArray(),
+            custom_extension_count = customExtensionCount,
+        };
+    }
+
+    private static void GetExtensions(CinemachineVirtualCameraBase vcamBase, out List<string> vcamExtensions, out int customExtensionCount)
+    {
+        customExtensionCount = 0;
+        // collect extensions on vcam
+        vcamExtensions = new List<string>();
+        var extensions = vcamBase.mExtensions;
+        if (extensions != null)
+        {
+            foreach (var extension in extensions)
+            {
+                string extensionName = extension.GetType().ToString();
+                switch (extensionName)
+                {
+                    case "CinemachineCameraOffset":
+                    case "Cinemachine3rdPersonAim":
+                    case "CinemachineCollider":
+                    case "CinemachineConfiner":
+                    case "CinemachineConfiner2D":
+                    case "CinemachineFollowZoom":
+                    case "CinemachineRecomposer":
+                    case "CinemachineStoryboard":
+                        break;
+                    default:
+                        extensionName = k_customStr; // hide users extension's name
+                        customExtensionCount++;
+                        break;
+                }
+                vcamExtensions.Add(extensionName);
+            }
+        }
+    }
+
+    private static string GetVcamClassName(CinemachineVirtualCameraBase vcamBase)
+    {
+        var type = vcamBase.GetType().ToString();
+        return type.Contains("Cinemachine.") ? type.Substring("Cinemachine.".Length) : k_customStr;
```

Comment:
inefficient to be constantly recomputing Substring("Cinemachine.".Length).  Just hardcode the number, with a comment